### PR TITLE
Fixing environment checker format and login

### PR DESCRIPTION
**Pull request by RazTamir**
Signed-off-by: ratamir <ratamir@redhat.com>

### CODE REVIEW

**Comment by RazTamir** [comment]
Verified:

https://paste.fedoraproject.org/paste/V~8-TizZfFOi~9sG2fsEeQ

Also tested killing pods to check those are not considered as added/removed resources

**Comment by vasukulkarni** [comment]
Since we are here, can we add option to skip this when doing script testing, It allows to focus on fixing script issues first when new development work is in progress(and not see huge logs) and then eventually fix the left overs.

**Comment by RazTamir** [comment]
Hi @vasukulkarni ,
I'm not sure I want to have that option.
It can introduce a lot of issues when people forget to enable it again and then an entire tier level can be regressed.
It is annoying, yes, but necessary.

In case we'll agree in the future to allow that flexibility, we can handle it

**Comment by vasukulkarni** [comment]
@RazTamir yeah agree there is a window to miss the testing with flag enabled, I think if we enable jenkins based PR testing this can be solved. we can visit this later on. I think I have an issue filed on this one.

File: ocs_ci/utility/environment_check.py [hunk 1]
```diff
@@ -85,73 +118,58 @@ def get_status_after_execution():
...
-    namespace_diff = DeepDiff(
+    namespace_diff = compare_dicts(
         ENV_STATUS_PRE['namespace'], ENV_STATUS_POST['namespace']
     )
     diffs_dict = {
         'pods': pod_diff,
-        'sc': sc_diff,
+        'storageClasses': sc_diff,
         'cephfs': cephfs_diff,
         'cephbp': cephbp_diff,
```
Comment (vasukulkarni) [inline ocs_ci/utility/environment_check.py]: Isn't this cephblockpool?

Comment (RazTamir) [reply ocs_ci/utility/environment_check.py]: cephblockpool is 2 lines below (cephbp).
It is just a name for the resource

**Comment by RazTamir** [comment]
Thanks.
Final verification process through Jenkins is our next (one of many :) ) thing to leverage

File: ocs_ci/utility/environment_check.py [hunk 2]
```diff
@@ -38,11 +37,45 @@
...
 
 
-ADDED_RESOURCE = 'iterable_item_added'
-REMOVED_RESOURCE = 'iterable_item_removed'
-
-
-def assign_get_values(env_status_dict, key, kind):
+def compare_dicts(before, after):
+    """
+    Comparing 2 dicts and providing diff list of [added items, removed items]
```
[the remark this answers is not in the document]
Comment (RazTamir) [reply ocs_ci/utility/environment_check.py]: Done

File: ocs_ci/utility/environment_check.py [hunk 3]
```diff
@@ -5,8 +5,7 @@
 import copy
 import logging
```
[the remark this answers is not in the document]
Comment (RazTamir) [reply ocs_ci/utility/environment_check.py]: done

File: ocs_ci/utility/environment_check.py [hunk 4]
```diff
@@ -5,8 +5,7 @@
...
 
-from deepdiff import DeepDiff
-import pprint
+import yaml
 from gevent.threadpool import ThreadPoolExecutor
```
[the remark this answers is not in the document]
Comment (RazTamir) [reply ocs_ci/utility/environment_check.py]: done

File: ocs_ci/utility/environment_check.py [hunk 5]
```diff
@@ -38,11 +37,47 @@
...
+    Comparing 2 dicts and providing diff list of [added items, removed items]
 
+    Args:
+        before (dict): Dictionary before execution
+        after (dict): Dictionary after execution
 
-def assign_get_values(env_status_dict, key, kind):
+    Returns:
+        list: List of 2 dicts - [added items, removed items]
+    """
```
[the remark this answers is not in the document]
Comment (RazTamir) [reply ocs_ci/utility/environment_check.py]: This is ok to have:
```python
"""
description ...

Args:
    arg1 (): abc
"""
```

File: ocs_ci/utility/environment_check.py [hunk 6]
```diff
@@ -38,11 +37,47 @@
...
+    Comparing 2 dicts and providing diff list of [added items, removed items]
 
+    Args:
+        before (dict): Dictionary before execution
+        after (dict): Dictionary after execution
 
-def assign_get_values(env_status_dict, key, kind):
+    Returns:
+        list: List of 2 dicts - [added items, removed items]
```
[the remark this answers is not in the document]
Comment (RazTamir) [reply ocs_ci/utility/environment_check.py]: Done

File: ocs_ci/utility/environment_check.py [hunk 7]
```diff
@@ -85,73 +118,58 @@ def get_status_after_execution():
...
-                    ]
-                })
-            except KeyError:
-                leftovers['Leftovers added'].append({
-                    kind: kind_diff[ADDED_RESOURCE]
-                })
+        if kind_diff[0]:
+            leftovers[
+                'Leftovers added'
+            ].append({f"***{kind}***": kind_diff[0]})
```
Comment (fbalak) [inline ocs_ci/utility/environment_check.py]: I would drop `*` symbols so it wouldn't look like this:
```
E           Resources added:
E           - '***pods***':
E             - apiVersion: v1
E               kind: Pod
```
But more like:
```
E           Resources added:
E           - pods:
E             - apiVersion: v1
E               kind: Pod
```
It feels more consistent with `yaml` format for me but that's just my opinion.

File: ocs_ci/utility/environment_check.py [hunk 8]
```diff
@@ -85,73 +118,58 @@ def get_status_after_execution():
...
-                    ]
-                })
-            except KeyError:
-                leftovers['Leftovers removed'].append({
-                    kind: kind_diff[REMOVED_RESOURCE]
-                })
+        if kind_diff[1]:
+            leftovers[
+                'Leftovers removed'
+            ].append({f"***{kind}***": kind_diff[1]})
```
Comment (fbalak) [inline ocs_ci/utility/environment_check.py]: same

File: ocs_ci/utility/environment_check.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (vavuthu) [reply ocs_ci/utility/environment_check.py]: its good to have,,I think we incorporated that from google sphinx-doc https://www.sphinx-doc.org/en/1.5/ext/example_google.html

Its not a show stopper

File: ocs_ci/utility/environment_check.py (hunk 7 above)
Comment (RazTamir) [reply ocs_ci/utility/environment_check.py]: I got feedback from ppl that when the output is long, they can't seem to understand what the resource leftovers it

File: ocs_ci/utility/environment_check.py (hunk 8 above)
Comment (RazTamir) [reply ocs_ci/utility/environment_check.py]: same

File: ocs_ci/utility/environment_check.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (RazTamir) [reply ocs_ci/utility/environment_check.py]: That's good to know - thanks!
We'll need to update the documentation but please let's not block this PR

**Comment by shylesh** [comment]
@RazTamir 
Can you paste a sample output, just to check how it looks now.

**Comment by RazTamir** [comment]
E           There are leftovers in the environment after test case:
E           Resources added:
E           - '***pvs***':
E             - apiVersion: v1
E               kind: PersistentVolume
E               metadata:
E                 annotations:
E                   pv.kubernetes.io/provisioned-by: rbd.csi.ceph.com
E                 creationTimestamp: '2019-07-31T16:13:11Z'
E                 finalizers:
E                 - kubernetes.io/pv-protection
E                 name: pvc-15dd7b5e-b3ae-11e9-a5fd-06e1ab615f34
E                 resourceVersion: '25211'
E                 selfLink: /api/v1/persistentvolumes/pvc-15dd7b5e-b3ae-11e9-a5fd-06e1ab615f34
E                 uid: 17838314-b3ae-11e9-b261-02b6255b949e
E               spec:
E                 accessModes:
E                 - ReadWriteOnce
E                 capacity:
E                   storage: 40Gi
E                 claimRef:
E                   apiVersion: v1
E                   kind: PersistentVolumeClaim
E                   name: my-alertmanager-claim-alertmanager-main-0
E                   namespace: openshift-monitoring
E                   resourceVersion: '24241'
E                   uid: 15dd7b5e-b3ae-11e9-a5fd-06e1ab615f34
E                 csi:
E                   driver: rbd.csi.ceph.com
E                   fsType: ext4
E                   nodeStageSecretRef:
E                     name: secret-test-rbd-3116123368
E                     namespace: rook-ceph
E                   volumeAttributes:
E                     clusterID: rook-ceph
E                     imageFeatures: layering
E                     imageFormat: '2'
E                     pool: cbp-test-3116123416
E                     storage.kubernetes.io/csiProvisionerIdentity: 1564588383894-8081-rbd.csi.ceph.com
E                   volumeHandle: 0001-0009-rook-ceph-0000000000000008-16301b80-b3ae-11e9-bd86-0a580a80020b
E                 persistentVolumeReclaimPolicy: Delete
E                 storageClassName: storageclass-test-rbd-3116124144
E                 volumeMode: Filesystem
E               status:
E                 phase: Released
E             - apiVersion: v1
E               kind: PersistentVolume
E               metadata:
E                 annotations:
E                   pv.kubernetes.io/provisioned-by: rbd.csi.ceph.com
E                 creationTimestamp: '2019-07-31T16:13:20Z'
E                 finalizers:
E                 - kubernetes.io/pv-protection
E                 name: pvc-1b57a199-b3ae-11e9-a5fd-06e1ab615f34
E                 resourceVersion: '25364'
E                 selfLink: /api/v1/persistentvolumes/pvc-1b57a199-b3ae-11e9-a5fd-06e1ab615f34
E                 uid: 1d0e3c9e-b3ae-11e9-b261-02b6255b949e
E               spec:
E                 accessModes:
E                 - ReadWriteOnce
E                 capacity:
E                   storage: 40Gi
E                 claimRef:
E                   apiVersion: v1
E                   kind: PersistentVolumeClaim
E                   name: my-prometheus-claim-prometheus-k8s-0
E                   namespace: openshift-monitoring
E                   resourceVersion: '24372'
E                   uid: 1b57a199-b3ae-11e9-a5fd-06e1ab615f34
E                 csi:
E                   driver: rbd.csi.ceph.com
E                   fsType: ext4
E                   nodeStageSecretRef:
E                     name: secret-test-rbd-3116123368
E                     namespace: rook-ceph
E                   volumeAttributes:
E                     clusterID: rook-ceph
E                     imageFeatures: layering
E                     imageFormat: '2'
E                     pool: cbp-test-3116123416
E                     storage.kubernetes.io/csiProvisionerIdentity: 1564588383894-8081-rbd.csi.ceph.com
E                   volumeHandle: 0001-0009-rook-ceph-0000000000000008-1b90f810-b3ae-11e9-bd86-0a580a80020b
E                 persistentVolumeReclaimPolicy: Delete
E                 storageClassName: storageclass-test-rbd-3116124144
E                 volumeMode: Filesystem
E               status:
E                 phase: Released
E             - apiVersion: v1
E               kind: PersistentVolume
E               metadata:
E                 annotations:
E                   pv.kubernetes.io/provisioned-by: rbd.csi.ceph.com
E                 creationTimestamp: '2019-07-31T16:13:20Z'
E                 finalizers:
E                 - kubernetes.io/pv-protection
E                 name: pvc-1b626876-b3ae-11e9-a5fd-06e1ab615f34
E                 resourceVersion: '25385'
E                 selfLink: /api/v1/persistentvolumes/pvc-1b626876-b3ae-11e9-a5fd-06e1ab615f34
E                 uid: 1d0f6810-b3ae-11e9-b261-02b6255b949e
E               spec:
E                 accessModes:
E                 - ReadWriteOnce
E                 capacity:
E                   storage: 40Gi
E                 claimRef:
E                   apiVersion: v1
E                   kind: PersistentVolumeClaim
E                   name: my-prometheus-claim-prometheus-k8s-1
E                   namespace: openshift-monitoring
E                   resourceVersion: '24382'
E                   uid: 1b626876-b3ae-11e9-a5fd-06e1ab615f34
E                 csi:
E                   driver: rbd.csi.ceph.com
E                   fsType: ext4
E                   nodeStageSecretRef:
E                     name: secret-test-rbd-3116123368
E                     namespace: rook-ceph
E                   volumeAttributes:
E                     clusterID: rook-ceph
E                     imageFeatures: layering
E                     imageFormat: '2'
E                     pool: cbp-test-3116123416
E                     storage.kubernetes.io/csiProvisionerIdentity: 1564588383894-8081-rbd.csi.ceph.com
E                   volumeHandle: 0001-0009-rook-ceph-0000000000000008-1b97628f-b3ae-11e9-bd86-0a580a80020b
E                 persistentVolumeReclaimPolicy: Delete
E                 storageClassName: storageclass-test-rbd-3116124144
E                 volumeMode: Filesystem
E               status:
E                 phase: Released
E             - apiVersion: v1
E               kind: PersistentVolume
E               metadata:
E                 annotations:
E                   pv.kubernetes.io/provisioned-by: rbd.csi.ceph.com
E                 creationTimestamp: '2019-07-31T16:13:30Z'
E                 finalizers:
E                 - kubernetes.io/pv-protection
E                 name: pvc-2262d495-b3ae-11e9-a5fd-06e1ab615f34
E                 resourceVersion: '25244'
E                 selfLink: /api/v1/persistentvolumes/pvc-2262d495-b3ae-11e9-a5fd-06e1ab615f34
E                 uid: 230d3301-b3ae-11e9-b261-02b6255b949e
E               spec:
E                 accessModes:
E                 - ReadWriteOnce
E                 capacity:
E                   storage: 40Gi
E                 claimRef:
E                   apiVersion: v1
E                   kind: PersistentVolumeClaim
E                   name: my-alertmanager-claim-alertmanager-main-1
E                   namespace: openshift-monitoring
E                   resourceVersion: '24493'
E                   uid: 2262d495-b3ae-11e9-a5fd-06e1ab615f34
E                 csi:
E                   driver: rbd.csi.ceph.com
E                   fsType: ext4
E                   nodeStageSecretRef:
E                     name: secret-test-rbd-3116123368
E                     namespace: rook-ceph
E                   volumeAttributes:
E                     clusterID: rook-ceph
E                     imageFeatures: layering
E                     imageFormat: '2'
E                     pool: cbp-test-3116123416
E                     storage.kubernetes.io/csiProvisionerIdentity: 1564588383894-8081-rbd.csi.ceph.com
E                   volumeHandle: 0001-0009-rook-ceph-0000000000000008-2289e5b1-b3ae-11e9-bd86-0a580a80020b
E                 persistentVolumeReclaimPolicy: Delete
E                 storageClassName: storageclass-test-rbd-3116124144
E                 volumeMode: Filesystem
E               status:
E                 phase: Released
E             - apiVersion: v1
E               kind: PersistentVolume
E               metadata:
E                 annotations:
E                   pv.kubernetes.io/provisioned-by: rbd.csi.ceph.com
E                 creationTimestamp: '2019-07-31T16:14:01Z'
E                 finalizers:
E                 - kubernetes.io/pv-protection
E                 name: pvc-34a009a7-b3ae-11e9-a5fd-06e1ab615f34
E                 resourceVersion: '25249'
E                 selfLink: /api/v1/persistentvolumes/pvc-34a009a7-b3ae-11e9-a5fd-06e1ab615f34
E                 uid: 3554dc08-b3ae-11e9-b261-02b6255b949e
E               spec:
E                 accessModes:
E                 - ReadWriteOnce
E                 capacity:
E                   storage: 40Gi
E                 claimRef:
E                   apiVersion: v1
E                   kind: PersistentVolumeClaim
E                   name: my-alertmanager-claim-alertmanager-main-2
E                   namespace: openshift-monitoring
E                   resourceVersion: '24740'
E                   uid: 34a009a7-b3ae-11e9-a5fd-06e1ab615f34
E                 csi:
E                   driver: rbd.csi.ceph.com
E                   fsType: ext4
E                   nodeStageSecretRef:
E                     name: secret-test-rbd-3116123368
E                     namespace: rook-ceph
E                   volumeAttributes:
E                     clusterID: rook-ceph
E                     imageFeatures: layering
E                     imageFormat: '2'
E                     pool: cbp-test-3116123416
E                     storage.kubernetes.io/csiProvisionerIdentity: 1564588383894-8081-rbd.csi.ceph.com
E                   volumeHandle: 0001-0009-rook-ceph-0000000000000008-34cc229a-b3ae-11e9-bd86-0a580a80020b
E                 persistentVolumeReclaimPolicy: Delete
E                 storageClassName: storageclass-test-rbd-3116124144
E                 volumeMode: Filesystem
E               status:
E                 phase: Released
E           
E           Resources removed:
E            []

**Comment by shylesh** [comment]
Looks much better than earlier.

**Pull request merged**